### PR TITLE
Upload `bin` directory to azure artifacts

### PR DESCRIPTION
**Pull request by ShortDevelopment**
## Goals
- Provide "nightly" builds for all platforms
- Allow to debug per PR issues using the binaries from a CI-Run

## Issues
- macOS-builds created on CirrusCI are not included

---

Docs: [Azure Pipeline Artifacts](https://learn.microsoft.com/en-us/azure/devops/pipelines/artifacts/pipeline-artifacts?view=azure-devops&tabs=yaml#publish-artifacts)

### CODE REVIEW

File: azure-pipelines.yml
Comment (ShortDevelopment) [inline azure-pipelines.yml]: NoJit-MSVC builds use `.\Build\VcBuild.NoJIT\bin` as output directory!

**Comment by ShortDevelopment** [comment]
@ppenzin Could you have a quick look at this?

**Comment by rhuanjl** [comment]
@ppenzin what do you think of this temporarily storing our CI builds?

It's not clear how long they kept for.

**Comment by ShortDevelopment** [comment]
@rhuanjl This can be configured in the Azure DevOps settings
https://learn.microsoft.com/en-us/azure/devops/pipelines/policies/retention

**Review by JoseIvanHernandez — APPROVED** [review]
Commit

**Review by JoseIvanHernandez — APPROVED** [review]
Commit

**Review by JoseIvanHernandez — APPROVED** [review]
Commit

**Review by JoseIvanHernandez — APPROVED** [review]
Commit

**Review by JoseIvanHernandez — APPROVED** [review]
Commit

**Review by JoseIvanHernandez — APPROVED** [review]
Commit

**Review by JoseIvanHernandez — APPROVED** [review]
Commit

**Comment by ShortDevelopment** [comment]
@JoseIvanHernandez Please stop the spam! 
I've already reported you to GitHub!

**Comment by JoseIvanHernandez** [comment]
Apologies. I'm new and didn't realize it was spamming you. Certainly was not my intention. You're the first to reach out, I am very sorry.

**Comment by rhuanjl** [comment]
> Apologies. I'm new and didn't realize it was spamming you. Certainly was not my intention. You're the first to reach out, I am very sorry. 

When you comment on a pull request everyone tagged in it gets a notification.

I'm a little confused what you were trying to do here; a pull request (PR) is a request for the maintainers of a project to merge a contribution of some kind. This project is currently maintained by @ppenzin and I; it's up to us if/when a PR is merged.

This PR is a suggested developer experience feature from @ShortDevelopment, we're waiting until my fellow maintainer @ppenzin has a chance to consider it before deciding if to merge it.

If you have any other questions about this project (chakra core) or how to contribute to it feel free to either open an issue or join our discord server to chat (see the read me on the code page for a discord link).

**Review by ppenzin — APPROVED** [review]
This should work, I think Azure purges the storage periodically and _shouldn't_ get overwhelmed, but we can adjust the frequency of publishing later.

**Comment by ShortDevelopment** [comment]
I've done a lot of small commits
Just squash-merge the whole thing

**Pull request merged**
